### PR TITLE
chore: update golangci-lint

### PR DESCRIPTION
**Pull request by AlekSi**
Fix context propagation.

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4532)

### CODE REVIEW

File: pkg/machinery/client/events.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ func (c *Client) EventsWatch(ctx context.Context, watchFunc func(<-chan Event),
 		return err
 	}
 
-	defaultNode := RemotePeer(stream.Context())
+	defaultNode := RemotePeer(stream.Context()) //nolint:contextcheck
```
Comment (AlekSi) [inline pkg/machinery/client/events.go:79]: Seems to be the only real false positive

File: internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go [hunk 2]
```diff
@@ -455,7 +455,8 @@ func (s *Server) Upgrade(ctx context.Context, in *machine.UpgradeRequest) (reply
 				defer mu.Unlock(ctx) //nolint:errcheck
 			}
 
-			if err := s.Controller.Run(context.Background(), runtime.SequenceStageUpgrade, in); err != nil {
+			// ???
+			if err := s.Controller.Run(ctx, runtime.SequenceStageUpgrade, in); err != nil {
```
[the remark this answers is not in the document]
Comment (smira) [reply internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go:459]: this does `context.Background()` on purpose - basically API call finishes immediately, but the actual upgrade runs in the background (when API call finishes). 

(same below)

**Review by smira — APPROVED** [review]

**Pull request merged**